Python/flowgraph.c

```diff
@@ -1567,6 +1567,33 @@ apply_static_swaps(basicblock *block, int i)
     }
 }
 
+// Attempt to get the next non-NOP instruction from the current instruction
+// inst at position i in basic block bb. Return 1 if successful or 0 if
+// running out of blocks.
+static int
+next_instruction(basicblock **bb, cfg_instr **inst, int *i)
+{
+    ++*inst;
+    ++*i;
```

Comment:
This function probably isn't quite right. If the current instruction is a jump then then b_next is not the next block executed.  It also ignores line numbers - sometimes a NOP is there because it represents a line of code that needs to show as "executed" in tracing or coverage. We can't just skip over such NOPs (you should have a test where the ifexpr spans multiple lines).

I think rather than skip over NOPs here, we may need to add a call to ``remove_redundant_nops`` somewhere (maybe at the end of inline_small_or_no_lineno_blocks, only when changes > 0?).